content/ui-patterns/forms.mdx

```diff
@@ -0,0 +1,393 @@
+
+Placeholder text is never an acceptable substitute for a label because:
+
+- The placeholder text disappears as soon as the input has a value
+- Placeholder text colors are typically too light to meet the minimum color contrast ratio required for accessibility
+- Screen readers do not read placeholder text
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065187-d300024d-ce39-4f5e-82ad-9e74c4b6d28a.png" />
+    <Caption>Use a placeholder to hint at what kind of information is expected</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065188-35462873-3706-4063-b7b9-9d0993665307.png" />
+    <Caption>Don’t use a placeholder as a substitute for a label</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Required field indicator
+
+When a field is required to have a value, it should be visibly marked as required. An individual checkbox or radio button cannot be marked as required.
+
+For more information, see the [required fields](#required-fields) section.
+
+### Input (required)
+
+The input is what the user interacts with to set the value of the form control. You may pre-fill inputs with smart default values, but be careful about making too many assumptions about what a user wants or needs.
+
+### Caption
+
+A caption may be used to provide additional context about the field to help users fill in the correct data or explain how the data will be used. Caption text should be as short as possible.
+
+Caption text may be displayed alongside a validation message, or it may be hidden if it only provides redundant information.
+
+Caption text may be used to augment the [label](#label), but should not be redundant with the label or any other parts of the form control. If the caption feels redundant, try removing it. 
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/166832266-c34634ec-2ade-4bf6-8401-df0bdc02b324.png" />
+    <Caption>Use caption text that shows new information and provides helpful context</Caption>
+  </Do>
+  <Dont>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/166832267-e00f88bb-6a81-42e2-a29d-63cc58a70550.png" />
+    <Caption>Don’t use caption text that is redundant</Caption>
+  </Dont>
+</DoDontContainer>
+
+### Validation message
+
+A single validation message may be displayed to provide helpful information for a user to complete their task. For example: explain why a value is invalid so they can correct it and submit the form.
+
+An individual checkbox or radio should not have its own validation.
+
+<DoDontContainer>
+  <Do>
+    <img role="presentation" src="https://user-images.githubusercontent.com/2313998/164065196-b0146ade-eccb-4a71-b1b6-f4a82b96c115.png" />
+    <Caption>Show a validation message for the group of inputs</Caption>
```

Comment:
Have we considered a validation pattern where if there are multiple errors, we render a single box above the form with a list of each error- which links to each input with the issue? Its my understanding that this is the preferred method for handling multiple form errors, but perhaps that case hasn't come up for accessibility testing? 

This is from the book [Form Design Patterns](https://www.smashingmagazine.com/printed-books/form-design-patterns/)

![image from Form Design Patterns book showing an error box above a group of fields, linking to each field that is invalid](https://user-images.githubusercontent.com/18661030/167961570-bcc840ee-e4f0-4fb0-adf4-43d383384625.png)

Reply:
I had considered this pattern, but I tried it on a settings page and it felt like too much error text was being shown at once.

However, this pattern is useful and I think it's worth further exploration. Do you think we could explore it in a separate issue? Or should I do it as part of this PR?

Reply:
@mperrotti have you explored this?

Reply:
>  ![image](https://user-images.githubusercontent.com/293280/169881879-b9e9e92f-696c-4205-8275-e3c428d4f020.png)

@mperrotti additionally, is this "Pick your top two" and "Only two selections are allowed" a good recommendation for checkboxes? It feels out of place — I could see that pattern happening for a larger number ("Pick up to 10 items"?), but two...? Feels like a weird pattern...

Reply:
I thought I removed this example. If it's still around somewhere, I'll change it.

Reply:
I see this is still used used for a do/don't example.

<img width="977" alt="Screen Shot 2022-05-23 at 5 59 28 PM" src="https://user-images.githubusercontent.com/2313998/169912361-6a1bce1c-8f72-42a4-adc1-bda234dc168d.png">